plasmapy/particles/particle_class.py

```diff
@@ -234,10 +234,10 @@ class Particle(AbstractPhysicalParticle):
         integer representing the atomic number of an element; or a
         |Particle| instance.
 
-    mass_numb : `int`, optional
+    mass_numb : `int`, optional, keyword-only
         The mass number of an isotope or nuclide.
 
-    Z : `int`, optional
+    Z : `int`, optional, keyword-only
```

Comment:
I'm not sure how I feel about using `keyword-only` here.  I have never seen any documentation use this term, or practice.  From Python to `numpy` to `sphinx` and beyond, no one calls out keyword-only arguments.

I accept the use of `optional` only because it is [documented by `numpydoc`](https://numpydoc.readthedocs.io/en/latest/format.html#parameters).  Additionally, `optional` implies `keyword-only`.

I'm not convinced this is a practice we should pickup, but I'm open to hearing arguments for it.

Reply:
I may have originally learned this from Astropy's documentation (e.g., [`Supergalactic`](https://docs.astropy.org/en/stable/api/astropy.coordinates.builtin_frames.Supergalactic.html)).  Also, an argument can be keyword-only and not optional, or optional and not keyword-only.
```Python
def keyword_only_and_required(x, *, y): ...
def optional_and_not_keyword_only(x, y=None): ...
```
I think it's useful to specify that it's keyword-only for the reasons that explicit is better than implicit.  Eventually it might be useful to have "keyword-only" automatically link to the corresponding term in our glossary.  I also noticed that there was a discussion on https://github.com/numpy/numpydoc/issues/330 about this.